test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -654,7 +654,7 @@ void overwriteBlockHeaderForTest(mObject const& _blObj, TestBlock& _block, Chain
             ho.count("number") ? toU256(ho["number"]) : header.number(),
             ho.count("gasLimit") ? toU256(ho["gasLimit"]) : header.gasLimit(),
             ho.count("gasUsed") ? toU256(ho["gasUsed"]) : header.gasUsed(),
-            ho.count("timestamp") ? toU256(ho["timestamp"]) : header.timestamp(),
+            ho.count("timestamp") ? toU256(ho["timestamp"]) : header.timestamp() >= 0 ? header.timestamp() : 1000,
```

Comment:
What's wrong with timestamp 0?

Reply:
header.timestamp() could be negative which is incorrect and breaks the test generation.

Reply:
If it's not RLP parsing as Pawel suggests, maybe you should change `TestBlock` so that it produces a valid header.

Reply:
perhaps in NoProof::generateSeal ?

Reply:
nope that does not work

Reply:
So I have a question then. if blockheader by default is invalid. how I make it valid?

Reply:
Any seal engine should not seal an invalid block nor transform an invalid one into a valid one.

To make it valid set the timestamp proper value (a number >= 0).

Reply:
that exactly what I do at line 657

Reply:
It's `invalid` because it is invalid by default. what number should I use?

Reply:
0?

Reply:
testeth transform blockheader before mining for testing purposes.

Reply:
0 and utcTime() both works.